apps/api/src/app/step-schemas/e2e/get-step-schema.e2e.ts

```diff
@@ -0,0 +1,283 @@
+    it('should get step schema for existing step with previous steps', async function () {
+      const { data } = (
+        await session.testAgent.get(
+          `/v1/step-schemas?workflowId=${createdWorkflow._id}&stepId=${createdWorkflow.steps[1].stepUuid}`
+        )
+      ).body;
+
+      expect(data.variables.properties.steps.type).to.equal('object');
+      const variableStepKeys = Object.keys(data.variables.properties.steps.properties);
+      expect(variableStepKeys).to.have.length(1);
+      const variableStepKey = variableStepKeys[0];
+      const createdWorkflowPreviousSteps = createdWorkflow.steps.slice(
+        0,
+        createdWorkflow.steps.findIndex((stepItem) => stepItem.stepUuid === createdWorkflow.steps[1].stepUuid)
+      );
+      const variableStepKeyFoundInCreatedWorkflow = createdWorkflowPreviousSteps.find(
+        (step) => step.stepUuid === variableStepKey
+      );
+      const isValidVariableStepKey = !!variableStepKeyFoundInCreatedWorkflow;
+      expect(isValidVariableStepKey).to.be.true;
+      expect(data.variables.properties.steps.properties).to.have.property(variableStepKey);
+      expect(data.variables.properties.steps.properties[variableStepKey].type).to.equal('object');
+      expect(data.variables.properties.steps.properties[variableStepKey].properties).to.have.all.keys(
+        'seen',
+        'read',
+        'lastSeenDate',
+        'lastReadDate'
+      );
+      expect(data.variables.properties.steps.properties[variableStepKey].required).to.deep.equal([
+        'seen',
+        'read',
+        'lastSeenDate',
+        'lastReadDate',
+      ]);
+      expect(data.variables.properties.steps.properties[variableStepKey].additionalProperties).to.be.false;
+      expect(data.variables.properties.steps.required).to.be.an('array').that.is.empty;
+      expect(data.variables.properties.steps.additionalProperties).to.be.false;
+      expect(data.variables.properties.steps.description).to.equal('Previous Steps Results');
+    });
+
+    it('should get error for invalid step id', async function () {
+      const invalidStepUuid = `${createdWorkflow.steps[0].stepUuid}0`;
+
+      const res = await session.testAgent.get(
+        `/v1/step-schemas?workflowId=${createdWorkflow._id}&stepId=${invalidStepUuid}`
+      );
+      expect(res.status).to.equal(400);
+      expect(res.body.message).to.equal(`No step found with the given id ${invalidStepUuid}`);
+      expect(res.body.error).to.equal('Bad Request');
+      expect(res.body.statusCode).to.equal(400);
+    });
+
+    it('should get error for invalid workflow id', async function () {
+      const invalidWorkflowId = createdWorkflow.steps[0].stepUuid;
+
+      const res = await session.testAgent.get(
+        `/v1/step-schemas?workflowId=${invalidWorkflowId}&stepId=${createdWorkflow.steps[0].stepUuid}`
+      );
+      expect(res.status).to.equal(400);
+      expect(res.body.message).to.equal(`No workflow found with the given id ${invalidWorkflowId}`);
```

Comment:
We previously agreed that error messages should not be parameterized, and this implementation seems to not meet that standard.

Reply:
You are right!!!!! it was a bad habit!